distributed/src/main/scala/asobu/distributed/service/DRequestExtractor.scala

```diff
@@ -0,0 +1,154 @@
+package asobu.distributed.service
+
+import asobu.distributed.service.extractors.DRequestExtractor
+import asobu.distributed.protocol.{RequestParams, DRequest}
+import asobu.distributed.service.DRequestExtractor.{RequestParamsExtractor, BodyExtractor}
+import asobu.dsl._
+import asobu.dsl.extractors.JsonBodyExtractor
+import asobu.dsl.util.HListOps.{Combine3To, RestOf, CombineTo, RestOf2}
+import cats.{Monad, Functor, Eval}
+import cats.sequence.RecordSequencer
+import shapeless.ops.hlist.Prepend
+import asobu.dsl.util.RecordOps.{FieldKV, FieldKVs}
+import cats.data.{Xor, Kleisli}
+import play.api.libs.json.{JsError, JsSuccess, Reads, Json}
+import play.core.routing.RouteParams
+import shapeless.labelled.FieldType
+import shapeless.ops.hlist.Mapper
+import shapeless._, labelled.field
+import ExtractResult._
+import cats.syntax.all._
+import CatsInstances._
+import play.api.mvc._, play.api.mvc.Results._
+import Extractor._
+
+import scala.annotation.implicitNotFound
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.implicitConversions
+import scala.util.{Failure, Success, Try}
+
+object DRequestExtractor extends ExtractorFunctions {
```

Comment:
Just out of curiosity, do we have any idea on how much overhead this introduces?  This is pretty "hot" code, and executes on every single request.